Enable REST API by default in template configuration

TomaszGaweda opened the pull request:
Enable REST API by default in template configuration, so that e.g. `hz-healthcheck` script can use it.

Checklist:
- [x] Labels (`Team:`, `Type:`, `Source:`, `Module:`) and Milestone set
- [x] Label `Add to Release Notes` or `Not Release Notes content` set
- [x] Request reviewers if possible
- [x] Send backports/forwardports if fix needs to be applied to past/future releases
- [x] New public APIs have `@Nonnull/@Nullable` annotations
- [x] New public APIs have `@since` tags in Javadoc

kwart commented on hazelcast/src/main/config-template/hazelcast-assembly.xml:
```diff
@@ -174,7 +174,7 @@
         <!--
             Configuration for REST endpoints. When it's enabled, it will allow healthcheck calls over the HTTP protocol.
```
```suggestion
            Configuration for REST endpoints. It allows healthcheck calls over the HTTP protocol.
```

kwart commented on hazelcast/src/main/config-template/hazelcast-assembly.yaml:
```diff
@@ -113,7 +113,7 @@ hazelcast:
         enabled: false
     # Configuration for REST endpoints. When it's enabled, it will allow healthcheck calls over the HTTP protocol.
```
```suggestion
    # Configuration for REST endpoints. It allows healthcheck calls over the HTTP protocol.
```

hz-devops-test commented:
The job `Hazelcast-pr-EE-compiler` of your PR failed. (Hazelcast internal details: [build log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-EE-compiler/2434/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-EE-compiler/2434/Hazelcast-pr-EE-compiler-2434.zip)).
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] COMPILATION ERROR : 
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:compile (default-compile) on project hazelcast-enterprise: Compilation failure: Compilation failure: 
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/replication/WanBatchPublisher.java:[437,4] error: method does not override or implement a method from a supertype
--------------------------
</pre>
</details>

kwart reviewed (approved):
Approving and suggesting also comments update.

ldziedziul reviewed (commented):
I think we should update this as well: https://github.com/hazelcast/hazelcast/blob/6d681870b005e166b17ef7cdc5794bda9d704c27/hazelcast/src/main/resources/hazelcast-config-5.2.json#L731

TomaszGaweda commented:
> I think we should update this as well:
> 
> https://github.com/hazelcast/hazelcast/blob/6d681870b005e166b17ef7cdc5794bda9d704c27/hazelcast/src/main/resources/hazelcast-config-5.2.json#L731

True, updated as well, also xml config for 5.2

hz-devops-test commented:
The job `Hazelcast-pr-EE-compiler` of your PR failed. (Hazelcast internal details: [build log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-EE-compiler/2436/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-EE-compiler/2436/Hazelcast-pr-EE-compiler-2436.zip)).
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] COMPILATION ERROR : 
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:compile (default-compile) on project hazelcast-enterprise: Compilation failure: Compilation failure: 
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/replication/WanBatchPublisher.java:[437,4] error: method does not override or implement a method from a supertype
--------------------------
</pre>
</details>

hz-devops-test commented:
The job `Hazelcast-pr-EE-compiler` of your PR failed. (Hazelcast internal details: [build log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-EE-compiler/2437/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-EE-compiler/2437/Hazelcast-pr-EE-compiler-2437.zip)).
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] COMPILATION ERROR : 
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:compile (default-compile) on project hazelcast-enterprise: Compilation failure: Compilation failure: 
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/replication/WanBatchPublisher.java:[437,4] error: method does not override or implement a method from a supertype
--------------------------
</pre>
</details>

kwart commented:
> > I think we should update this as well:
> > https://github.com/hazelcast/hazelcast/blob/6d681870b005e166b17ef7cdc5794bda9d704c27/hazelcast/src/main/resources/hazelcast-config-5.2.json#L731
> 
> True, updated as well, also xml config for 5.2

This is not correct, the default value is `false`. We explicitly set it to `true` in some distributions. If the value is not specified, the REST has to remain disabled.

kwart reviewed (changes requested):
> > I think we should update this as well:
> > https://github.com/hazelcast/hazelcast/blob/6d681870b005e166b17ef7cdc5794bda9d704c27/hazelcast/src/main/resources/hazelcast-config-5.2.json#L731
> 
> True, updated as well, also xml config for 5.2

This is not correct, the default value is `false`. We explicitly set it to `true` in some distributions. If the value is not specified, the REST has to remain disabled.

TomaszGaweda commented:
> > > I think we should update this as well:
> > > https://github.com/hazelcast/hazelcast/blob/6d681870b005e166b17ef7cdc5794bda9d704c27/hazelcast/src/main/resources/hazelcast-config-5.2.json#L731
> > 
> > 
> > True, updated as well, also xml config for 5.2
> 
> This is not correct, the default value is `false`. We explicitly set it to `true` in some distributions. If the value is not specified, the REST has to remain disabled.

Thanks for clarification, updated :)

hz-devops-test commented:
The job `Hazelcast-pr-EE-compiler` of your PR failed. (Hazelcast internal details: [build log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-EE-compiler/2439/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-EE-compiler/2439/Hazelcast-pr-EE-compiler-2439.zip)).
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] COMPILATION ERROR : 
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:compile (default-compile) on project hazelcast-enterprise: Compilation failure: Compilation failure: 
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/replication/WanBatchPublisher.java:[437,4] error: method does not override or implement a method from a supertype
--------------------------
</pre>
</details>

hz-devops-test commented:
The job `Hazelcast-pr-EE-compiler` of your PR failed. (Hazelcast internal details: [build log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-EE-compiler/2441/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-EE-compiler/2441/Hazelcast-pr-EE-compiler-2441.zip)).
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] COMPILATION ERROR : 
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:compile (default-compile) on project hazelcast-enterprise: Compilation failure: Compilation failure: 
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler_2/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/replication/WanBatchPublisher.java:[437,4] error: method does not override or implement a method from a supertype
--------------------------
</pre>
</details>

TomaszGaweda commented:
run-lab-run

hz-devops-test commented:
The job `Hazelcast-pr-builder` of your PR failed. (Hazelcast internal details: [build log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/13212/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/13212/Hazelcast-pr-builder-13212.zip)).
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   BackupTest.testBackupRemoveWhenOwnerNodeDead:344 ? TestTimedOut test timed out...
[INFO] 
[ERROR] Tests run: 50896, Failures: 0, Errors: 1, Skipped: 238
[INFO] 

[ERROR] There are test failures.
</pre>
</details>

hz-devops-test commented:
The job `Hazelcast-pr-EE-compiler` of your PR failed. (Hazelcast internal details: [build log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-EE-compiler/2449/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-EE-compiler/2449/Hazelcast-pr-EE-compiler-2449.zip)).
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
---------SUMMARY----------
--------------------------
[ERROR] COMPILATION ERROR : 
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-compiler-plugin:2.5.1:compile (default-compile) on project hazelcast-enterprise: Compilation failure: Compilation failure: 
--------------------------
---------ERRORS-----------
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/EnterpriseWanReplicationService.java:[328,46] error: method publishAntiEntropyEvent in interface InternalWanPublisher<T> cannot be applied to given types;
--------------------------
[ERROR] /home/jenkins/jenkins_slave/workspace/Hazelcast-pr-EE-compiler/hazelcast-enterprise/hazelcast-enterprise/src/main/java/com/hazelcast/enterprise/wan/impl/replication/WanBatchPublisher.java:[437,4] error: method does not override or implement a method from a supertype
--------------------------
</pre>
</details>

hz-devops-test commented:
The job `Hazelcast-pr-builder` of your PR failed. (Hazelcast internal details: [build log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/13214/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/13214/Hazelcast-pr-builder-13214.zip)).
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   BasicNestedFieldsTest.test_typeCoercionUpserts:360->execute:576->execute:580 ? HazelcastSql
[INFO] 
[ERROR] Tests run: 5115, Failures: 0, Errors: 1, Skipped: 9
[INFO] 

[ERROR] There are test failures.
</pre>
</details>

hz-devops-test commented:
The job `Hazelcast-pr-builder` of your PR failed. (Hazelcast internal details: [build log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/13215/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/13215/Hazelcast-pr-builder-13215.zip)).
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Errors: 
[ERROR]   BasicNestedFieldsTest.test_typeCoercionUpserts:360->execute:576->execute:580 ? HazelcastSql
[INFO] 
[ERROR] Tests run: 5115, Failures: 0, Errors: 1, Skipped: 9
[INFO] 

[ERROR] There are test failures.
</pre>
</details>

hz-devops-test commented:
The job `Hazelcast-pr-builder` of your PR failed. (Hazelcast internal details: [build log](
https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/13224/build_log.txt), [artifacts](https://j-artifacts.s3.amazonaws.com/Hazelcast-pr-builder/13224/Hazelcast-pr-builder-13224.zip)).
Through arcane magic we have determined that the following fragments from the build log may contain information about the problem.
<details>
<summary>Click to expand the log file</summary>
<pre>
--------------------------
-------TEST FAILURE-------
--------------------------
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   ProtocolNegotiationTest.verifyOnlyTheProtocolHeaderIsSent:93 expected java.lang.IllegalStateException to be thrown, but nothing was thrown
[INFO] 
[ERROR] Tests run: 4327, Failures: 1, Errors: 0, Skipped: 8
[INFO] 

[ERROR] There are test failures.
</pre>
</details>

TomaszGaweda commented:
run-lab-run
